### DISCUSSION TITLE
Linking errors on WIndows due to symbols from the FMT dependency being already defined

### DISCUSSION THREAD

**Discussion by merwaaan** [OpenImageIO/oiio#3284] [Q&A] [answered]
Hi. Is anybody having this particular issue?

Our project uses OIIO, which depends on FMT and bundles it.
**Our project uses another library (spdlog) that also depends on FMT.**

OIIO and the other library both include FMT's headers so some of FMT's symbols end up in their respective .lib files.
When building our own project, which links against those two, we get linking errors because some symbols from fmt are already defined:

```
2>fmtd.lib(format.obj) : error LNK2005: "void __cdecl fmt::v8::detail::assert_fail(char const *,int,char const *)" (?assert_fail@detail@v8@fmt@@YAXPEBDH0@Z) already defined in OurProject.lib(utilities.obj)
2>fmtd.lib(format.obj) : error LNK2005: "public: void __cdecl fmt::v8::detail::error_handler::on_error(char const *)" (?on_error@error_handler@detail@v8@fmt@@QEAAXPEBD@Z) already defined in OurProject.lib(utilities.obj)
2>fmtd.lib(format.obj) : error LNK2005: "class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > __cdecl fmt::v8::vformat(class fmt::v8::basic_string_view<char>,class fmt::v8::basic_format_args<class fmt::v8::basic_format_context<class fmt::v8::appender,char> >)" (?vformat@v8@fmt@@YA?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@V?$basic_string_view@D@12@V?$basic_format_args@V?$basic_format_context@Vappender@v8@fmt@@D@v8@fmt@@@12@@Z) already defined in OurProject.lib(Generation.obj)
```

Not really unexpected since FMT's headers contain some templated code that ends up in both OpenImageIO.lib and our otherlib.lib.

Any idea how to handle this?

**Comment by lgritz** [comment, marked as answer]
I'm thinking that the problem is either that you're mixing versions, or that you are getting two copies of the library.

Do you build OIIO yourself? If fmt is installed somewhere, you ought to be able to use fmt_ROOT to direct OIIO to build against the very same fmt that your application is using, and maybe that will alleviate the conflict and/or duplication?

**Reply by merwaaan** [reply]
> I'm thinking that the problem is either that you're mixing versions

That was it. I thought that we were using our own build of FMT for both OIIO and the other dependency, but OIIO actually downloaded its own mismatching version due to an incorrect configuration on our part.

Thanks!
